diff --git a/da_vinci.theme b/da_vinci.theme
index f27d053..d47a877 100644
--- a/da_vinci.theme
+++ b/da_vinci.theme
@@ -275,10 +275,12 @@ function da_vinci_theme_suggestions_form_alter(array &$suggestions, array $varia
 function da_vinci_theme_suggestions_file_link_alter(array &$suggestions, array $variables) {
   // Get the node type.
   if ($node = \Drupal::routeMatch()->getParameter('node')) {
-    $type = $node->getType();
+    if (!empty($node) && is_object($node)) {
+      $type = $node->getType();

-    // Create the theme hook suggestion.
-    $suggestions[] = 'file_link__type_' . $type;
+      // Create the theme hook suggestion.
+      $suggestions[] = 'file_link__type_' . $type;
+    }
   }
 }
